### DISCUSSION TITLE
How to apply for SPECT-CT image

### DISCUSSION THREAD

**Discussion by akmal871026** [villekf/OMEGA#6] [General]
Hi All,

If I want to reconstruction image for SPECT raw Dicom, is it possible?

**Comment by villekf** [comment]
SPECT support is not available at the moment. SPECT support might be added in the last half of next year though.

**Comment by akmal871026** [comment]
Dear Villekf/Omega

Can i pay you to you build the coding for spect ct image reconstruction?

You tell me how much?

Please  consider it sir. Because i have to send my project as soon as
possible.

**Reply by villekf** [reply]
Unfortunately I do not have the time at the moment. In case you haven't tried them yet, you could try using QSPECT, STIR or NiftyRec.

**Comment by villekf** [comment]
The current version allows the reconstruction of SPECT data, potentially also from DICOM data. There is an example of Siemens Pro.specta scanner with DICOM data, though at the moment the data is not publicly available. That will, hopefully, change later this month.

**Reply by villekf** [reply]
Just an update on the data availability that unfortunately no DICOM data can be made available and currently only the Simind example data is available.

**Comment by akmal871026** [comment]
Ok Dr. Thanks you very much

**Comment by akmal871026** [comment]
Dear @villekf ,

I got error like below:

Warning: ROOT support is only supported when compiling with Visual Studio! 
> In install_mex (line 640) 
Warning: Unable to build ROOT support! If you do not need ROOT support ignore this warning. Otherwise,
make sure that ROOT was compiled with the same compiler you are using to compile this mex-file and
provide the path to ROOT install folder with install_mex(0, [], [], [], 'C:/path/to/ROOT'). Compiler
error is shown with install_mex(1) 
> In install_mex (line 667) 
Warning: No OpenCL SDK found. Implementations 2 and 3 will not be built. Use install_mex(1,
'C:/PATH/TO/OPENCL/INCLUDE', 'C:/PATH/TO/OPENCL/LIB/X64') to set OpenCL include and library paths. 
> In install_mex (line 711) 



do you have any ways to solve?

**Reply by villekf** [reply]
If you want OpenCL support, you need to have OpenCL installed. Did you look at: https://omega-doc.readthedocs.io/en/latest/installation.html already?

**Comment by akmal871026** [comment]
I installed OpenCL. But still error


*___________________________________*

*MOHD AKMAL BIN MASUD*

H/P: +60143321631

**Reply by villekf** [reply]
Then I recommend you just use the CPU implementation (i.e. implementation 4). You can ignore all those previous errors then.

**Comment by akmal871026** [comment]
I try run matlab in linux. Then let you know. Give me some time

**Comment by akmal871026** [comment]
Sir,

I have a problem below.

What is the problem, sir?

>> install_mex
Warning: No OpenCL SDK detected. If one is installed, insert the paths
manually by using install_mex(0,
'C:\PATH\TO\OPENCL\INCLUDE', 'C:\PATH\TO\OPENCL\LIB\X64')
> In install_mex (line 397)
Building with 'Microsoft Visual C++ 2022'.
MEX completed successfully.
Building with 'Microsoft Visual C++ 2022'.
MEX completed successfully.
Implementations 1 & 4 built with OpenMP (parallel) support
Building with 'Microsoft Visual C++ 2022'.
MEX completed successfully.
Building with 'Microsoft Visual C++ 2022'.
MEX completed successfully.
Building with 'Microsoft Visual C++ 2022'.
MEX completed successfully.
Building with 'Microsoft Visual C++ 2022'.
MEX completed successfully.
Inveon support enabled
Warning: Unable to build ROOT support! If you do not need ROOT support
ignore this warning. Otherwise, make sure that ROOT
was compiled with the same compiler you are using to compile this mex-file
and provide the path to ROOT install folder with
install_mex(0, [], [], [], 'C:/path/to/ROOT'). Compiler error is shown with
install_mex(1)
> In install_mex (line 667)
Warning: No OpenCL SDK found. Implementations 2 and 3 will not be built.
Use install_mex(1, 'C:/PATH/TO/OPENCL/INCLUDE',
'C:/PATH/TO/OPENCL/LIB/X64') to set OpenCL include and library paths.
> In install_mex (line 711)


*___________________________________*

*MOHD AKMAL BIN MASUD*

H/P: +60143321631

**Reply by villekf** [reply]
If you don't intend to use OpenCL support, there is no problem. If you intend to use OpenCL, you need to install it. If you use a Nvidia GPU, then CUDA toolkit is fine. For SPECT, you don't really need OpenCL support though so everything is fine.
